src/actions.ml

```diff
@@ -257,6 +257,55 @@ let send_doc_url ~bot_info ~github_repo_full_name job_info =
   | _ ->
       Lwt.return_unit
 
+let send_bench_url ~bot_info job_info (gh_owner, gh_repo) ~external_id =
+  Lwt_io.print "Pushing status check for bench job.\n"
+  <&>
+  let url = f "https://gitlab.com/coq/coq/-/jobs/%d" job_info.build_id in
+  let state = job_info.build_status in
+  let context = "bench" in
+  url |> Uri.of_string |> Client.get
+  >>= fun (resp, _) ->
+  let status_code = resp |> Response.status |> Code.code_of_status in
+  if Int.equal 200 status_code then
+    let state, status, conclusion, description =
+      match state with
+      | "success" ->
+          ("success", COMPLETED, Some SUCCESS, "Bench completed successfully")
+      | "created" ->
+          ( "pending"
+          , QUEUED
+          , None
+          , "Bench ready — Comment \"@coqbot bench\" to start bench" )
+      | "running" ->
+          ("pending", IN_PROGRESS, None, "Bench in progress")
+      | "cancelled" | "canceled" ->
+          ("failure", COMPLETED, Some CANCELLED, "Bench cancelled")
+      | "failed" ->
+          ("failure", COMPLETED, Some NEUTRAL, "Bench completed with failures")
+      | _ ->
+          failwith (f "Error: send_bench_url received unknown state %s." state)
+    in
+    let description = description ^ "." in
+    match bot_info.github_install_token with
+    | None ->
+        GitHub_mutations.send_status_check ~repo_full_name:"coq/coq"
+          ~commit:job_info.common_info.head_commit ~state ~url ~context
+          ~description ~bot_info
+    | Some _ -> (
+        GitHub_queries.get_repository_id ~bot_info ~owner:gh_owner ~repo:gh_repo
+        >>= function
+        | Ok repo_id ->
+            GitHub_mutations.create_check_run ~bot_info ~name:context ~status
+              ~repo_id ~head_sha:job_info.common_info.head_commit ?conclusion
+              ~title:description ~details_url:url ~summary:"" ~external_id ()
+        | Error e ->
+            Lwt_io.printf "No repo id: %s\n" e )
```

Comment:
You can consider that for the Coq repository, we are always in the `Some` case. So the `get_repository_id` request could be factorized at the beginning, then you would call `create_check_run` directly for each type of change.

Reply:
Removed all of it now.